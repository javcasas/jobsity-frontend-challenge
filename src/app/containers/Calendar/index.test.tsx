import * as React from 'react';
import { Calendar, Props } from './index';
import { shallow, mount } from 'enzyme';
import { utc } from 'moment';
import { spy } from 'sinon';

describe("Calendar container", () => {
  const defaultProps : Props = {
    reminders: [],
    date: utc("2019-04-14T12:11:45"),
    createReminder: () => {},
    updateReminder: () => {},
  }

  function mkShallow(props?: Props) {
    const finalProps = {...defaultProps, ...props}
    return shallow(<Calendar {...finalProps} />)
  }
  function mkDeep(props?: Props) {
    const finalProps = {...defaultProps, ...props}
    return mount(<Calendar {...finalProps} />)
  }

  it("render calendar", () => {
    expect(mkShallow()).toMatchSnapshot();
  })
  it("Create new reminder button opens the form", () => {
    const calendar = mkShallow();
    calendar.find("button.create-reminder").simulate("click");
    expect(calendar).toMatchSnapshot();
  })
  it("Create new reminder button opens the form", () => {
    const calendar = mkShallow();
    calendar.find("button.create-reminder").simulate("click");
    expect(calendar).toMatchSnapshot();
  })
  it("Create new reminder - full workflow", () => {
    const createReminder = spy();
    const calendar = mkDeep({...defaultProps, createReminder});
    calendar.find("button.create-reminder").simulate("click");
    expect(calendar).toMatchSnapshot();
    calendar.find("ReminderForm").setState({text: "asdf", city: "London"})
    expect(calendar).toMatchSnapshot();
    expect(calendar.find("form div button[disabled=false]:first-child").length).toBe(1);
    calendar.find("form div button:first-child").simulate("click");
    expect(createReminder.calledOnce).toBe(true);
    expect(createReminder.firstCall.args).toStrictEqual([{
      city: "London",
      color: 0,
      date: utc("2019-04-14T12:11:45"),
      id: 0,
      text: "asdf"
    }]);
  })
})
