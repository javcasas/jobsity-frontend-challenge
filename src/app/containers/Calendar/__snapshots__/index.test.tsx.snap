// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar container Create new reminder - full workflow 1`] = `
<Calendar
  createReminder={[Function]}
  date={"2019-04-14T12:11:45.000Z"}
  reminders={Array []}
  updateReminder={[Function]}
>
  <ReminderForm
    initialDate={"2019-04-14T12:11:45.000Z"}
    onCancel={[Function]}
    onCreate={[Function]}
    saveText="Save new reminder"
    title="Create new reminder"
  >
    <div>
      <form>
        <h1>
          Create new reminder
        </h1>
        <label
          htmlFor="description"
        >
          Description
        </label>
        <input
          id="description"
          onChange={[Function]}
          placeholder="Up to 30 chars of description"
          type="text"
          value=""
        />
        <ErrorMsg>
          <span>
            Please enter a description
          </span>
        </ErrorMsg>
        <label
          htmlFor="city"
        >
          City where the even will happen
        </label>
        <input
          id="city"
          onChange={[Function]}
          type="text"
          value=""
        />
        <ErrorMsg>
          <span>
            Please enter a city
          </span>
        </ErrorMsg>
        <label
          htmlFor="date"
        >
          Date
        </label>
        <input
          id="date"
          onChange={[Function]}
          type="date"
          value="2019-04-14"
        />
        <label
          htmlFor="time"
        >
          Time
        </label>
        <input
          id="time"
          onChange={[Function]}
          type="time"
          value="12:11"
        />
        <label
          htmlFor="red"
        >
          <input
            checked={true}
            id="red"
            onChange={[Function]}
            type="radio"
          />
          Red
        </label>
        <label
          htmlFor="green"
        >
          <input
            checked={false}
            id="green"
            onChange={[Function]}
            type="radio"
          />
          Green
        </label>
        <label
          htmlFor="blue"
        >
          <input
            checked={false}
            id="blue"
            onChange={[Function]}
            type="radio"
          />
          Blue
        </label>
        <label
          htmlFor="yellow"
        >
          <input
            checked={false}
            id="yellow"
            onChange={[Function]}
            type="radio"
          />
          Yellow
        </label>
        <div>
          <button
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Save new reminder
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </ReminderForm>
  <Month
    elements={Array []}
    today={"2019-04-14T12:11:45.000Z"}
  >
    <h1>
      April 2019
    </h1>
    <table>
      <thead>
        <tr>
          <th>
            Sunday
          </th>
          <th>
            Monday
          </th>
          <th>
            Tuesday
          </th>
          <th>
            Wednesday
          </th>
          <th>
            Thursday
          </th>
          <th>
            Friday
          </th>
          <th>
            Saturday
          </th>
        </tr>
      </thead>
      <tbody>
        <Component
          elements={Array []}
          firstDay={"2019-03-31T00:00:00.000Z"}
          key="2019-03-31T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-03-31T00:00:00.000Z"}
              isToday={false}
              key="2019-03-31T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                31
              </td>
            </Component>
            <Component
              date={"2019-04-01T00:00:00.000Z"}
              isToday={false}
              key="2019-04-01T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                1
              </td>
            </Component>
            <Component
              date={"2019-04-02T00:00:00.000Z"}
              isToday={false}
              key="2019-04-02T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                2
              </td>
            </Component>
            <Component
              date={"2019-04-03T00:00:00.000Z"}
              isToday={false}
              key="2019-04-03T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                3
              </td>
            </Component>
            <Component
              date={"2019-04-04T00:00:00.000Z"}
              isToday={false}
              key="2019-04-04T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                4
              </td>
            </Component>
            <Component
              date={"2019-04-05T00:00:00.000Z"}
              isToday={false}
              key="2019-04-05T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                5
              </td>
            </Component>
            <Component
              date={"2019-04-06T00:00:00.000Z"}
              isToday={false}
              key="2019-04-06T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                6
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-07T00:00:00.000Z"}
          key="2019-04-07T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-07T00:00:00.000Z"}
              isToday={false}
              key="2019-04-07T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                7
              </td>
            </Component>
            <Component
              date={"2019-04-08T00:00:00.000Z"}
              isToday={false}
              key="2019-04-08T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                8
              </td>
            </Component>
            <Component
              date={"2019-04-09T00:00:00.000Z"}
              isToday={false}
              key="2019-04-09T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                9
              </td>
            </Component>
            <Component
              date={"2019-04-10T00:00:00.000Z"}
              isToday={false}
              key="2019-04-10T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                10
              </td>
            </Component>
            <Component
              date={"2019-04-11T00:00:00.000Z"}
              isToday={false}
              key="2019-04-11T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                11
              </td>
            </Component>
            <Component
              date={"2019-04-12T00:00:00.000Z"}
              isToday={false}
              key="2019-04-12T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                12
              </td>
            </Component>
            <Component
              date={"2019-04-13T00:00:00.000Z"}
              isToday={false}
              key="2019-04-13T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                13
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-14T00:00:00.000Z"}
          key="2019-04-14T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-14T00:00:00.000Z"}
              isToday={true}
              key="2019-04-14T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className="undefined"
              >
                14
              </td>
            </Component>
            <Component
              date={"2019-04-15T00:00:00.000Z"}
              isToday={false}
              key="2019-04-15T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                15
              </td>
            </Component>
            <Component
              date={"2019-04-16T00:00:00.000Z"}
              isToday={false}
              key="2019-04-16T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                16
              </td>
            </Component>
            <Component
              date={"2019-04-17T00:00:00.000Z"}
              isToday={false}
              key="2019-04-17T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                17
              </td>
            </Component>
            <Component
              date={"2019-04-18T00:00:00.000Z"}
              isToday={false}
              key="2019-04-18T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                18
              </td>
            </Component>
            <Component
              date={"2019-04-19T00:00:00.000Z"}
              isToday={false}
              key="2019-04-19T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                19
              </td>
            </Component>
            <Component
              date={"2019-04-20T00:00:00.000Z"}
              isToday={false}
              key="2019-04-20T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                20
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-21T00:00:00.000Z"}
          key="2019-04-21T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-21T00:00:00.000Z"}
              isToday={false}
              key="2019-04-21T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                21
              </td>
            </Component>
            <Component
              date={"2019-04-22T00:00:00.000Z"}
              isToday={false}
              key="2019-04-22T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                22
              </td>
            </Component>
            <Component
              date={"2019-04-23T00:00:00.000Z"}
              isToday={false}
              key="2019-04-23T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                23
              </td>
            </Component>
            <Component
              date={"2019-04-24T00:00:00.000Z"}
              isToday={false}
              key="2019-04-24T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                24
              </td>
            </Component>
            <Component
              date={"2019-04-25T00:00:00.000Z"}
              isToday={false}
              key="2019-04-25T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                25
              </td>
            </Component>
            <Component
              date={"2019-04-26T00:00:00.000Z"}
              isToday={false}
              key="2019-04-26T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                26
              </td>
            </Component>
            <Component
              date={"2019-04-27T00:00:00.000Z"}
              isToday={false}
              key="2019-04-27T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                27
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-28T00:00:00.000Z"}
          key="2019-04-28T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-28T00:00:00.000Z"}
              isToday={false}
              key="2019-04-28T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                28
              </td>
            </Component>
            <Component
              date={"2019-04-29T00:00:00.000Z"}
              isToday={false}
              key="2019-04-29T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                29
              </td>
            </Component>
            <Component
              date={"2019-04-30T00:00:00.000Z"}
              isToday={false}
              key="2019-04-30T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                30
              </td>
            </Component>
            <Component
              date={"2019-05-01T00:00:00.000Z"}
              isToday={false}
              key="2019-05-01T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                1
              </td>
            </Component>
            <Component
              date={"2019-05-02T00:00:00.000Z"}
              isToday={false}
              key="2019-05-02T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                2
              </td>
            </Component>
            <Component
              date={"2019-05-03T00:00:00.000Z"}
              isToday={false}
              key="2019-05-03T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                3
              </td>
            </Component>
            <Component
              date={"2019-05-04T00:00:00.000Z"}
              isToday={false}
              key="2019-05-04T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                4
              </td>
            </Component>
          </tr>
        </Component>
      </tbody>
    </table>
  </Month>
  <button
    className="create-reminder"
    onClick={[Function]}
  >
    Create new reminder
  </button>
  <label
    htmlFor="api_key"
  >
    API key
  </label>
  <input
    id="api_key"
    onChange={[Function]}
    type="text"
  />
</Calendar>
`;

exports[`Calendar container Create new reminder - full workflow 2`] = `
<Calendar
  createReminder={[Function]}
  date={"2019-04-14T12:11:45.000Z"}
  reminders={Array []}
  updateReminder={[Function]}
>
  <ReminderForm
    initialDate={"2019-04-14T12:11:45.000Z"}
    onCancel={[Function]}
    onCreate={[Function]}
    saveText="Save new reminder"
    title="Create new reminder"
  >
    <div>
      <form>
        <h1>
          Create new reminder
        </h1>
        <label
          htmlFor="description"
        >
          Description
        </label>
        <input
          id="description"
          onChange={[Function]}
          placeholder="Up to 30 chars of description"
          type="text"
          value="asdf"
        />
        <label
          htmlFor="city"
        >
          City where the even will happen
        </label>
        <input
          id="city"
          onChange={[Function]}
          type="text"
          value="London"
        />
        <label
          htmlFor="date"
        >
          Date
        </label>
        <input
          id="date"
          onChange={[Function]}
          type="date"
          value="2019-04-14"
        />
        <label
          htmlFor="time"
        >
          Time
        </label>
        <input
          id="time"
          onChange={[Function]}
          type="time"
          value="12:11"
        />
        <label
          htmlFor="red"
        >
          <input
            checked={true}
            id="red"
            onChange={[Function]}
            type="radio"
          />
          Red
        </label>
        <label
          htmlFor="green"
        >
          <input
            checked={false}
            id="green"
            onChange={[Function]}
            type="radio"
          />
          Green
        </label>
        <label
          htmlFor="blue"
        >
          <input
            checked={false}
            id="blue"
            onChange={[Function]}
            type="radio"
          />
          Blue
        </label>
        <label
          htmlFor="yellow"
        >
          <input
            checked={false}
            id="yellow"
            onChange={[Function]}
            type="radio"
          />
          Yellow
        </label>
        <div>
          <button
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Save new reminder
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </ReminderForm>
  <Month
    elements={Array []}
    today={"2019-04-14T12:11:45.000Z"}
  >
    <h1>
      April 2019
    </h1>
    <table>
      <thead>
        <tr>
          <th>
            Sunday
          </th>
          <th>
            Monday
          </th>
          <th>
            Tuesday
          </th>
          <th>
            Wednesday
          </th>
          <th>
            Thursday
          </th>
          <th>
            Friday
          </th>
          <th>
            Saturday
          </th>
        </tr>
      </thead>
      <tbody>
        <Component
          elements={Array []}
          firstDay={"2019-03-31T00:00:00.000Z"}
          key="2019-03-31T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-03-31T00:00:00.000Z"}
              isToday={false}
              key="2019-03-31T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                31
              </td>
            </Component>
            <Component
              date={"2019-04-01T00:00:00.000Z"}
              isToday={false}
              key="2019-04-01T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                1
              </td>
            </Component>
            <Component
              date={"2019-04-02T00:00:00.000Z"}
              isToday={false}
              key="2019-04-02T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                2
              </td>
            </Component>
            <Component
              date={"2019-04-03T00:00:00.000Z"}
              isToday={false}
              key="2019-04-03T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                3
              </td>
            </Component>
            <Component
              date={"2019-04-04T00:00:00.000Z"}
              isToday={false}
              key="2019-04-04T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                4
              </td>
            </Component>
            <Component
              date={"2019-04-05T00:00:00.000Z"}
              isToday={false}
              key="2019-04-05T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                5
              </td>
            </Component>
            <Component
              date={"2019-04-06T00:00:00.000Z"}
              isToday={false}
              key="2019-04-06T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                6
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-07T00:00:00.000Z"}
          key="2019-04-07T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-07T00:00:00.000Z"}
              isToday={false}
              key="2019-04-07T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                7
              </td>
            </Component>
            <Component
              date={"2019-04-08T00:00:00.000Z"}
              isToday={false}
              key="2019-04-08T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                8
              </td>
            </Component>
            <Component
              date={"2019-04-09T00:00:00.000Z"}
              isToday={false}
              key="2019-04-09T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                9
              </td>
            </Component>
            <Component
              date={"2019-04-10T00:00:00.000Z"}
              isToday={false}
              key="2019-04-10T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                10
              </td>
            </Component>
            <Component
              date={"2019-04-11T00:00:00.000Z"}
              isToday={false}
              key="2019-04-11T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                11
              </td>
            </Component>
            <Component
              date={"2019-04-12T00:00:00.000Z"}
              isToday={false}
              key="2019-04-12T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                12
              </td>
            </Component>
            <Component
              date={"2019-04-13T00:00:00.000Z"}
              isToday={false}
              key="2019-04-13T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                13
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-14T00:00:00.000Z"}
          key="2019-04-14T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-14T00:00:00.000Z"}
              isToday={true}
              key="2019-04-14T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className="undefined"
              >
                14
              </td>
            </Component>
            <Component
              date={"2019-04-15T00:00:00.000Z"}
              isToday={false}
              key="2019-04-15T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                15
              </td>
            </Component>
            <Component
              date={"2019-04-16T00:00:00.000Z"}
              isToday={false}
              key="2019-04-16T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                16
              </td>
            </Component>
            <Component
              date={"2019-04-17T00:00:00.000Z"}
              isToday={false}
              key="2019-04-17T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                17
              </td>
            </Component>
            <Component
              date={"2019-04-18T00:00:00.000Z"}
              isToday={false}
              key="2019-04-18T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                18
              </td>
            </Component>
            <Component
              date={"2019-04-19T00:00:00.000Z"}
              isToday={false}
              key="2019-04-19T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                19
              </td>
            </Component>
            <Component
              date={"2019-04-20T00:00:00.000Z"}
              isToday={false}
              key="2019-04-20T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                20
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-21T00:00:00.000Z"}
          key="2019-04-21T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-21T00:00:00.000Z"}
              isToday={false}
              key="2019-04-21T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                21
              </td>
            </Component>
            <Component
              date={"2019-04-22T00:00:00.000Z"}
              isToday={false}
              key="2019-04-22T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                22
              </td>
            </Component>
            <Component
              date={"2019-04-23T00:00:00.000Z"}
              isToday={false}
              key="2019-04-23T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                23
              </td>
            </Component>
            <Component
              date={"2019-04-24T00:00:00.000Z"}
              isToday={false}
              key="2019-04-24T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                24
              </td>
            </Component>
            <Component
              date={"2019-04-25T00:00:00.000Z"}
              isToday={false}
              key="2019-04-25T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                25
              </td>
            </Component>
            <Component
              date={"2019-04-26T00:00:00.000Z"}
              isToday={false}
              key="2019-04-26T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                26
              </td>
            </Component>
            <Component
              date={"2019-04-27T00:00:00.000Z"}
              isToday={false}
              key="2019-04-27T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                27
              </td>
            </Component>
          </tr>
        </Component>
        <Component
          elements={Array []}
          firstDay={"2019-04-28T00:00:00.000Z"}
          key="2019-04-28T00:00:00Z"
          today={"2019-04-14T12:11:45.000Z"}
        >
          <tr>
            <Component
              date={"2019-04-28T00:00:00.000Z"}
              isToday={false}
              key="2019-04-28T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                28
              </td>
            </Component>
            <Component
              date={"2019-04-29T00:00:00.000Z"}
              isToday={false}
              key="2019-04-29T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                29
              </td>
            </Component>
            <Component
              date={"2019-04-30T00:00:00.000Z"}
              isToday={false}
              key="2019-04-30T00:00:00Z"
              notThisMonth={false}
            >
              <td
                className=""
              >
                30
              </td>
            </Component>
            <Component
              date={"2019-05-01T00:00:00.000Z"}
              isToday={false}
              key="2019-05-01T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                1
              </td>
            </Component>
            <Component
              date={"2019-05-02T00:00:00.000Z"}
              isToday={false}
              key="2019-05-02T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                2
              </td>
            </Component>
            <Component
              date={"2019-05-03T00:00:00.000Z"}
              isToday={false}
              key="2019-05-03T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                3
              </td>
            </Component>
            <Component
              date={"2019-05-04T00:00:00.000Z"}
              isToday={false}
              key="2019-05-04T00:00:00Z"
              notThisMonth={true}
            >
              <td
                className=""
              >
                4
              </td>
            </Component>
          </tr>
        </Component>
      </tbody>
    </table>
  </Month>
  <button
    className="create-reminder"
    onClick={[Function]}
  >
    Create new reminder
  </button>
  <label
    htmlFor="api_key"
  >
    API key
  </label>
  <input
    id="api_key"
    onChange={[Function]}
    type="text"
  />
</Calendar>
`;

exports[`Calendar container Create new reminder button opens the form 1`] = `
<Fragment>
  <ReminderForm
    initialDate={"2019-04-14T12:11:45.000Z"}
    onCancel={[Function]}
    onCreate={[Function]}
    saveText="Save new reminder"
    title="Create new reminder"
  />
  <Month
    elements={Array []}
    today={"2019-04-14T12:11:45.000Z"}
  />
  <button
    className="create-reminder"
    onClick={[Function]}
  >
    Create new reminder
  </button>
  <label
    htmlFor="api_key"
  >
    API key
  </label>
  <input
    id="api_key"
    onChange={[Function]}
    type="text"
  />
</Fragment>
`;

exports[`Calendar container Create new reminder button opens the form 2`] = `
<Fragment>
  <ReminderForm
    initialDate={"2019-04-14T12:11:45.000Z"}
    onCancel={[Function]}
    onCreate={[Function]}
    saveText="Save new reminder"
    title="Create new reminder"
  />
  <Month
    elements={Array []}
    today={"2019-04-14T12:11:45.000Z"}
  />
  <button
    className="create-reminder"
    onClick={[Function]}
  >
    Create new reminder
  </button>
  <label
    htmlFor="api_key"
  >
    API key
  </label>
  <input
    id="api_key"
    onChange={[Function]}
    type="text"
  />
</Fragment>
`;

exports[`Calendar container render calendar 1`] = `
<Fragment>
  <Month
    elements={Array []}
    today={"2019-04-14T12:11:45.000Z"}
  />
  <button
    className="create-reminder"
    onClick={[Function]}
  >
    Create new reminder
  </button>
  <label
    htmlFor="api_key"
  >
    API key
  </label>
  <input
    id="api_key"
    onChange={[Function]}
    type="text"
  />
</Fragment>
`;
